src/doc_builder/utils.py

```diff
@@ -97,3 +106,71 @@ def get_doc_config():
     Returns the `doc_config` if it has been loaded.
     """
     return doc_config
+
+
+def is_doc_builder_repo(path):
+    """
+    Detects whether a folder is the `doc_builder` or not.
+    """
+    setup_file = Path(path) / "setup.py"
+    if not setup_file.exists():
+        return False
+    with open(os.path.join(path, "setup.py")) as f:
+        first_line = f.readline()
+    return first_line == "# Doc-builder package setup.\n"
+
+
+def locate_kit_folder():
+    """
+    Returns the location of the `kit` folder of `doc-builder`.
+
+    Will clone the doc-builder repo and cache it, if it's not found.
+    """
+    # First try: let's search where the module is.
+    repo_root = Path(__file__).parent.parent.parent
+    kit_folder = repo_root / "kit"
+    if kit_folder.is_dir():
+        return kit_folder
+
+    # Second try, maybe we are inside the doc-builder repo
+    current_dir = Path.cwd()
+    while current_dir.parent != current_dir and not (current_dir / ".git").is_dir():
+        current_dir = current_dir.parent
+    kit_folder = current_dir / "kit"
+    if kit_folder.is_dir() and is_doc_builder_repo(current_dir):
+        return kit_folder
+
+    # Otherwise, let's clone the repo and cache it.
+    return Path(get_cached_repo()) / "kit"
+
+
+def get_cached_repo():
+    """
+    Clone and cache the `doc-builder` repo.
+    """
+    os.makedirs(DOC_BUILDER_CACHE, exist_ok=True)
+    cache_repo_path = Path(DOC_BUILDER_CACHE) / "doc-builder-repo"
+    if not cache_repo_path.is_dir():
+        print(
+            "To build the HTML doc, we need the kit subfolder of the `doc-builder` repo. Cloning it and caching at "
+            f"{cache_repo_path}."
+        )
+        _ = subprocess.run(
+            "git clone https://github.com/huggingface/doc-builder.git".split(),
+            stderr=subprocess.PIPE,
+            stdout=subprocess.PIPE,
```

Comment:
These line will prevent logs from being output to the command line, won't logs be useful in some situations?